### PR TITLE
Strip and ignore fancier markdown link types and liquid tags

### LINKED ISSUE #9: Liquid tags should be ignored
e.g. `{% include ... %}`

### PR DESCRIPTION
**Pull request by jrha**
Including use of liquid tags for magic

Resolves #9, apart from comment blocks which are hard and IMHO should be spellchecked anyway.

### CODE REVIEW

**Comment by jouvin** [comment]
LGTM

File: src/testfile.md [hunk 1]
```diff
@@ -18,3 +18,16 @@ for err in self.spellcheck:
...
+
+{% comment %}
+this is a comment line
+or paragraph, it has to
+be spelled correctly
+{% endcomment %}
+
+[technical_fooblyness]: /wersdiem/technical_details.html#jkfsdl
+
+[meeseeks]: {% link _development/00-meeseeks.md %}
```
Comment (tofu-rocketry) [inline src/testfile.md:32]: I approve of this link ID name.

File: src/markspelling.py [hunk 2]
```diff
@@ -44,6 +44,9 @@ def __init__(self, pwl, language='en_GB', errortotalprev=0):
         self.regexhtmlclean = re.compile(r'\`.+?\`')
         self.regexlink = re.compile(r'\[(.+?)\]\(.+?\)')
         self.regexurl = re.compile(r'https?://\S+')
+        self.regexlinkid = re.compile(r'\[\S+\]:\s[^{\s]*')
+        self.regexlinkparam = re.compile(r'{:\S+="\S+"}')
+        self.regexliquidtag = re.compile(r'{%.+%}')
```
Comment (tofu-rocketry) [inline src/markspelling.py:49]: Do you only need to catch Liquid tags? What about objects (`{{ }}`)?

File: src/markspelling.py [hunk 3]
```diff
@@ -44,6 +44,9 @@ def __init__(self, pwl, language='en_GB', errortotalprev=0):
         self.regexhtmlclean = re.compile(r'\`.+?\`')
         self.regexlink = re.compile(r'\[(.+?)\]\(.+?\)')
```
Comment (tofu-rocketry) [inline src/markspelling.py:45]: There doesn't seem to be a test line for this.

**Review by tofu-rocketry — COMMENTED** [review]
Mostly looks good.

File: src/markspelling.py (hunk 3 above)
Comment (jrha) [reply src/markspelling.py:45]: There is now!

File: src/markspelling.py (hunk 2 above)
Comment (jrha) [reply src/markspelling.py:49]: Good point, added.

**Review by tofu-rocketry — APPROVED** [review]

**Pull request merged**